src/dmd/e2ir.d

```diff
@@ -2758,30 +2774,74 @@ elem *toElem(Expression e, IRState *irs)
 
                     assert(ae.e2.type.ty != Tpointer);
 
-                    if (!postblit && !destructor && !irs.arrayBoundsCheck())
+                    if (!postblit && !destructor)
                     {
                         elem *ex = el_same(&eto);
 
-                        // Determine if elen is a constant
-                        elem *elen;
-                        if (eto.Eoper == OPpair &&
-                            eto.EV.E1.Eoper == OPconst)
+                        /* Returns: length of array ex
+                         */
+                        static elem *getDotLength(elem *eto, elem *ex)
```

Comment:
Note: the logic of `getDotLength` is directly copied from the source, just moved to an inner function.